### ISSUE TITLE
Version 0.2 published in crates.io cannot be built anymore due to transitive dependency on quote 1.0.3

### ISSUE THREAD

**Issue by ancwrd1** [mullvad/windows-service-rs#43]
The `quote` crate was updated to 1.0.3 which broke older version of `err-derive` which is a dependency on published `windows-service 0.2`.

Using git repository directly as a dependency fixes the problem.

**Comment by nbigaouette** [comment]
`windows-service`'s commit e22e6d8281291a227b8f375fa13685be6ca8b146 has bumped the dependency on `err-derive` to 0.2.1, which should fix this.

Unfortunately, a new version was not released to crates.io (which is still at 0.2.0 from a year ago).

@faern Could a new release be made to crates.io? Until then we'll continue to need to add something like this to Cargo.toml:

```toml
[patch.crates-io]
windows-service = { git = 'https://github.com/mullvad/windows-service-rs.git', ref ="202d88bf438fdb870f4fc26e2031d36ab083fc42" }
```

**Comment by faern** [comment]
We can't release latest master of this repo. Because it depends on the git version of `winapi`, since `winapi` is stale and never releases anything in their master branch :man_shrugging:

**Comment by crusty-dave** [comment]
I just got bit by this as well, we are using our own mirror and this broke when we updated the mirro.

**Comment by crusty-dave** [comment]
Can windows-service please be modified to use err-derive "=0.1.5" instead of trying to go to the latest?

**Comment by crusty-dave** [comment]
For now, manually adding **err-derive** in my Cargo.toml is working for me:

```
[target.'cfg(windows)'.dependencies]
windows-service = {registry = "crates-io-mirror", version="0.2"}
winapi = {version = "0.3", features=["dpapi", "synchapi", "wincrypt"], registry = "crates-io-mirror"}
err-derive = {registry = "crates-io-mirror", version="=0.1.5"}
```

**Comment by patriksvensson** [comment]
Another workaround for this (since the crate isn't being updated), is by adding a package reference directly to the `windows-service-rs` repository.

```
windows-service = { git = "https://github.com/mullvad/windows-service-rs", rev="202d88bf438fdb870f4fc26e2031d36ab083fc42" }
```

**Comment by retep998** [comment]
Just ran into this issue myself. Could someone go poke the maintainer of `winapi` to push a new version to crates.io?

**Comment by Geobert** [comment]
> 
> 
> Just ran into this issue myself. Could someone go poke the maintainer of `winapi` to push a new version to crates.io?

I'm confused, aren't you the maintainer of `winapi`?

**Comment by faern** [comment]
Finally! We vendored the unpublished winapi bindings in #49. And now I published `windows-service 0.3.0`. I hope it solves this problem and that the new features are of great use to people.

**Comment by ancwrd1** [comment]
Thank you for the update.
